// terra-table/lib/terra-dev-site/doc/table/MultiSelectableRows.5.doc.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _terraDocTemplate = _interopRequireDefault(require("terra-doc-template"));

var _MultiSelectableRows = _interopRequireDefault(require("../../../../docs/MultiSelectableRows.md"));

var _package = require("../../../../package.json");

var _MultiSelectableRows2 = _interopRequireDefault(require("!terra-props-table-loader!../../../../src/MultiSelectableRows"));

var _MultiSelectableRows3 = _interopRequireDefault(require("!raw-loader!../../../../src/MultiSelectableRows"));

var _MultiSelectableRows4 = _interopRequireDefault(require("../example/MultiSelectableRows"));

var _MultiSelectableRows5 = _interopRequireDefault(require("!raw-loader!../../../../src/terra-dev-site/doc/example/MultiSelectableRows"));

var _MultiSelectableRowsMaxSelectable = _interopRequireDefault(require("../example/MultiSelectableRowsMaxSelectable"));

var _MultiSelectableRowsMaxSelectable2 = _interopRequireDefault(require("!raw-loader!../../../../src/terra-dev-site/doc/example/MultiSelectableRowsMaxSelectable"));

var _MultiSelectableRowsWithSubheader = _interopRequireDefault(require("../example/MultiSelectableRowsWithSubheader"));

var _MultiSelectableRowsWithSubheader2 = _interopRequireDefault(require("!raw-loader!../../../../src/terra-dev-site/doc/example/MultiSelectableRowsWithSubheader"));

var _MultiSelectableRowsOnChange = _interopRequireDefault(require("../example/MultiSelectableRowsOnChange"));

var _MultiSelectableRowsOnChange2 = _interopRequireDefault(require("!raw-loader!../../../../src/terra-dev-site/doc/example/MultiSelectableRowsOnChange"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// Example Files
var DocPage = function DocPage() {
  return _react.default.createElement(_terraDocTemplate.default, {
    packageName: _package.name,
    packageVersion: _package.version,
    readme: _MultiSelectableRows.default,
    srcPath: "https://github.com/cerner/terra-core/tree/master/packages/".concat(_package.name),
    examples: [{
      title: 'Multiple Selectable Rows',
      example: _react.default.createElement(_MultiSelectableRows4.default, null),
      source: _MultiSelectableRows5.default
    }, {
      title: 'Multiple Selectable Rows (Max selectable: 2)',
      example: _react.default.createElement(_MultiSelectableRowsMaxSelectable.default, null),
      source: _MultiSelectableRowsMaxSelectable2.default
    }, {
      title: 'Multiple Selectable Rows (Includes Subheaders)',
      example: _react.default.createElement(_MultiSelectableRowsWithSubheader.default, null),
      source: _MultiSelectableRowsWithSubheader2.default
    }, {
      title: 'Multiple Selectable Rows (onChange callback)',
      example: _react.default.createElement(_MultiSelectableRowsOnChange.default, null),
      source: _MultiSelectableRowsOnChange2.default
    }],
    propsTables: [{
      componentName: 'Multiple Selectable Rows',
      componentSrc: _MultiSelectableRows3.default,
      componentProps: _MultiSelectableRows2.default
    }]
  });
};

var _default = DocPage;
exports.default = _default;
