// terra-table/lib/terra-dev-site/doc/table/Rows.2.doc.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _terraDocTemplate = _interopRequireDefault(require("terra-doc-template"));

var _Rows = _interopRequireDefault(require("../../../../docs/Rows.md"));

var _package = require("../../../../package.json");

var _TableRows = _interopRequireDefault(require("!terra-props-table-loader!../../../../src/TableRows"));

var _TableRows2 = _interopRequireDefault(require("!raw-loader!../../../../src/TableRows"));

var _RowsOnClick = _interopRequireDefault(require("../example/RowsOnClick"));

var _RowsOnClick2 = _interopRequireDefault(require("!raw-loader!../../../../src/terra-dev-site/doc/example/RowsOnClick"));

var _RowsOnKeyDown = _interopRequireDefault(require("../example/RowsOnKeyDown"));

var _RowsOnKeyDown2 = _interopRequireDefault(require("!raw-loader!../../../../src/terra-dev-site/doc/example/RowsOnKeyDown"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DocPage = function DocPage() {
  return _react.default.createElement(_terraDocTemplate.default, {
    packageName: _package.name,
    packageVersion: _package.version,
    readme: _Rows.default,
    srcPath: "https://github.com/cerner/terra-core/tree/master/packages/".concat(_package.name),
    examples: [{
      title: 'Table Rows (onClick callback)',
      example: _react.default.createElement(_RowsOnClick.default, null),
      source: _RowsOnClick2.default
    }, {
      title: 'Table Rows (onKeyDown callback)',
      description: 'Note: Requires Table Row children to be focusable.',
      example: _react.default.createElement(_RowsOnKeyDown.default, null),
      source: _RowsOnKeyDown2.default
    }],
    propsTables: [{
      componentName: 'Table Rows',
      componentSrc: _TableRows2.default,
      componentProps: _TableRows.default
    }]
  });
};

var _default = DocPage;
exports.default = _default;
