// terra-table/lib/terra-dev-site/doc/table/SingleSelectableRows.4.doc.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _terraDocTemplate = _interopRequireDefault(require("terra-doc-template"));

var _SingleSelectableRows = _interopRequireDefault(require("../../../../docs/SingleSelectableRows.md"));

var _package = require("../../../../package.json");

var _SingleSelectableRows2 = _interopRequireDefault(require("!terra-props-table-loader!../../../../src/SingleSelectableRows"));

var _SingleSelectableRows3 = _interopRequireDefault(require("!raw-loader!../../../../src/SingleSelectableRows"));

var _SingleRowSelectableTable = _interopRequireDefault(require("../example/SingleRowSelectableTable"));

var _SingleRowSelectableTable2 = _interopRequireDefault(require("!raw-loader!../../../../src/terra-dev-site/doc/example/SingleRowSelectableTable"));

var _TableWithNonSelectableRow = _interopRequireDefault(require("../example/TableWithNonSelectableRow"));

var _TableWithNonSelectableRow2 = _interopRequireDefault(require("!raw-loader!../../../../src/terra-dev-site/doc/example/TableWithNonSelectableRow"));

var _SingleSelectableRowsOnChange = _interopRequireDefault(require("../example/SingleSelectableRowsOnChange"));

var _SingleSelectableRowsOnChangeAdditionalDetails = _interopRequireDefault(require("../example/SingleSelectableRowsOnChangeAdditionalDetails"));

var _SingleSelectableRowsOnChange2 = _interopRequireDefault(require("!raw-loader!../../../../src/terra-dev-site/doc/example/SingleSelectableRowsOnChange"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DocPage = function DocPage() {
  return _react.default.createElement(_terraDocTemplate.default, {
    packageName: _package.name,
    packageVersion: _package.version,
    readme: _SingleSelectableRows.default,
    srcPath: "https://github.com/cerner/terra-core/tree/master/packages/".concat(_package.name),
    examples: [{
      title: 'Single Selectable Rows',
      example: _react.default.createElement(_SingleRowSelectableTable.default, null),
      source: _SingleRowSelectableTable2.default
    }, {
      title: 'Single Selectable Rows (Second row unselectable)',
      example: _react.default.createElement(_TableWithNonSelectableRow.default, null),
      source: _TableWithNonSelectableRow2.default
    }, {
      title: 'Single Selectable Rows (onChange callback)',
      example: _react.default.createElement(_SingleSelectableRowsOnChange.default, null),
      source: _SingleSelectableRowsOnChange2.default
    }, {
      title: 'Single Selectable Rows (additional details from onChange)',
      example: _react.default.createElement(_SingleSelectableRowsOnChangeAdditionalDetails.default, null),
      source: _SingleSelectableRowsOnChange2.default
    }],
    propsTables: [{
      componentName: 'Single Selectable Rows',
      componentSrc: _SingleSelectableRows3.default,
      componentProps: _SingleSelectableRows2.default
    }]
  });
};

var _default = DocPage;
exports.default = _default;
