// terra-table/lib/terra-dev-site/doc/example/RowsOnKeyDown.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Table = _interopRequireDefault(require("terra-table/lib/Table"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

var RowsOnKeyDown =
/*#__PURE__*/
function (_React$Component) {
  _inherits(RowsOnKeyDown, _React$Component);

  function RowsOnKeyDown() {
    var _this;

    _classCallCheck(this, RowsOnKeyDown);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(RowsOnKeyDown).call(this));
    _this.onKeyDown = _this.onKeyDown.bind(_assertThisInitialized(_this));
    _this.clickedText = 'key pressed on body';
    return _this;
  }

  _createClass(RowsOnKeyDown, [{
    key: "onKeyDown",
    value: function onKeyDown() {
      // eslint-disable-next-line no-alert
      alert(this.clickedText);
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement(_Table.default, {
        isStriped: false
      }, _react.default.createElement(_Table.default.Header, null, _react.default.createElement(_Table.default.HeaderCell, {
        content: "Name",
        key: "NAME",
        minWidth: "small"
      }), _react.default.createElement(_Table.default.HeaderCell, {
        content: "Address",
        key: "ADDRESS",
        minWidth: "medium"
      }), _react.default.createElement(_Table.default.HeaderCell, {
        content: "Phone Number",
        key: "PHONE_NUMBER",
        minWidth: "large"
      })), _react.default.createElement(_Table.default.Rows, {
        onKeyDown: this.onKeyDown
      }, _react.default.createElement(_Table.default.Row, {
        key: "PERSON_0",
        tabIndex: "0"
      }, _react.default.createElement(_Table.default.Cell, {
        content: "John Smith",
        key: "NAME"
      }), _react.default.createElement(_Table.default.Cell, {
        content: "123 Adams Drive",
        key: "ADDRESS"
      }), _react.default.createElement(_Table.default.Cell, {
        content: "111-222-3333",
        key: "PHONE_NUMBER"
      })), _react.default.createElement(_Table.default.Row, {
        key: "PERSON_1",
        tabIndex: "0"
      }, _react.default.createElement(_Table.default.Cell, {
        content: "Jane Smith",
        key: "NAME"
      }), _react.default.createElement(_Table.default.Cell, {
        content: "321 Drive Street",
        key: "ADDRESS"
      }), _react.default.createElement(_Table.default.Cell, {
        content: "111-222-3333",
        key: "PHONE_NUMBER"
      })), _react.default.createElement(_Table.default.Row, {
        key: "PERSON_2",
        tabIndex: "0"
      }, _react.default.createElement(_Table.default.Cell, {
        content: "Dave Smith",
        key: "NAME"
      }), _react.default.createElement(_Table.default.Cell, {
        content: "213 Raymond Road",
        key: "ADDRESS"
      }), _react.default.createElement(_Table.default.Cell, {
        content: "111-222-3333",
        key: "PHONE_NUMBER"
      }))));
    }
  }]);

  return RowsOnKeyDown;
}(_react.default.Component);

var _default = RowsOnKeyDown;
exports.default = _default;
