// terra-table/lib/terra-dev-site/doc/example/SelectableRowsDisableUnselectedRows.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Table = _interopRequireDefault(require("terra-table/lib/Table"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var onChange = function onChange() {
  // eslint-disable-next-line no-alert
  alert('Attempted to change selected indexes.');
};

var SelectableRowsDisableUnselectedRows = function SelectableRowsDisableUnselectedRows() {
  return _react.default.createElement(_Table.default, {
    isStriped: false
  }, _react.default.createElement(_Table.default.Header, null, _react.default.createElement(_Table.default.HeaderCell, {
    content: "Name",
    key: "NAME",
    minWidth: "small"
  }), _react.default.createElement(_Table.default.HeaderCell, {
    content: "Address",
    key: "ADDRESS",
    minWidth: "medium"
  }), _react.default.createElement(_Table.default.HeaderCell, {
    content: "Phone Number",
    key: "PHONE_NUMBER",
    minWidth: "large"
  })), _react.default.createElement(_Table.default.SelectableTableRows, {
    selectedIndexes: [1],
    disableUnselectedRows: true,
    onChange: onChange
  }, _react.default.createElement(_Table.default.Row, {
    key: "PERSON_0"
  }, _react.default.createElement(_Table.default.Cell, {
    content: "John Smith",
    key: "NAME"
  }), _react.default.createElement(_Table.default.Cell, {
    content: "123 Adams Drive",
    key: "ADDRESS"
  }), _react.default.createElement(_Table.default.Cell, {
    content: "111-222-3333",
    key: "PHONE_NUMBER"
  })), _react.default.createElement(_Table.default.Row, {
    key: "PERSON_1"
  }, _react.default.createElement(_Table.default.Cell, {
    content: "Jane Smith",
    key: "NAME"
  }), _react.default.createElement(_Table.default.Cell, {
    content: "321 Drive Street",
    key: "ADDRESS"
  }), _react.default.createElement(_Table.default.Cell, {
    content: "111-222-3333",
    key: "PHONE_NUMBER"
  })), _react.default.createElement(_Table.default.Row, {
    key: "PERSON_2"
  }, _react.default.createElement(_Table.default.Cell, {
    content: "Dave Smith",
    key: "NAME"
  }), _react.default.createElement(_Table.default.Cell, {
    content: "213 Raymond Road",
    key: "ADDRESS"
  }), _react.default.createElement(_Table.default.Cell, {
    content: "111-222-3333",
    key: "PHONE_NUMBER"
  }))));
};

var _default = SelectableRowsDisableUnselectedRows;
exports.default = _default;
