// terra-table/lib/terra-dev-site/doc/example/SingleSelectableRowsOnChangeAdditionalDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Table = _interopRequireDefault(require("terra-table/lib/Table"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var people = [{
  key: 'PERSON_0',
  name: 'John Smith',
  address: '123 Adams Drive',
  phoneNumber: '111-222-3333'
}, {
  key: 'PERSON_1',
  name: 'Jane Smith',
  address: '321 Drive Street',
  phoneNumber: '111-222-3333'
}, {
  key: 'PERSON_2',
  name: 'Dave Smith',
  address: '213 Raymond Road',
  phoneNumber: '111-222-3333'
}];

var onChange = function onChange(event, selectedIndex) {
  // eslint-disable-next-line no-alert
  alert("Selected name: ".concat(people[selectedIndex].name));
};

var SingleSelectableRowsOnChangeAdditionalDetails = function SingleSelectableRowsOnChangeAdditionalDetails() {
  return _react.default.createElement(_Table.default, {
    isStriped: false
  }, _react.default.createElement(_Table.default.Header, null, _react.default.createElement(_Table.default.HeaderCell, {
    content: "Name",
    key: "NAME",
    minWidth: "small"
  }), _react.default.createElement(_Table.default.HeaderCell, {
    content: "Address",
    key: "ADDRESS",
    minWidth: "medium"
  }), _react.default.createElement(_Table.default.HeaderCell, {
    content: "Phone Number",
    key: "PHONE_NUMBER",
    minWidth: "large"
  })), _react.default.createElement(_Table.default.SingleSelectableRows, {
    onChange: onChange
  }, people.map(function (person) {
    return _react.default.createElement(_Table.default.Row, {
      key: person.key
    }, _react.default.createElement(_Table.default.Cell, {
      content: person.name,
      key: "NAME"
    }), _react.default.createElement(_Table.default.Cell, {
      content: person.address,
      key: "ADDRESS"
    }), _react.default.createElement(_Table.default.Cell, {
      content: person.phoneNumber,
      key: "PHONE_NUMBER"
    }));
  })));
};

var _default = SingleSelectableRowsOnChangeAdditionalDetails;
exports.default = _default;
