// terra-table/lib/terra-dev-site/doc/example/TableWithCustomCells.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Table = _interopRequireDefault(require("terra-table/lib/Table"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _TableWithCustomCellsModule = _interopRequireDefault(require("./TableWithCustomCells.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_TableWithCustomCellsModule.default);

var CustomCell = function CustomCell(props) {
  return _react.default.createElement("div", null, _react.default.createElement("h3", null, props.text), props.subtext ? _react.default.createElement("h4", {
    className: cx('custom-cell-header-wrapper')
  }, props.subtext) : null);
};

CustomCell.propTypes = {
  text: _propTypes.default.string,
  subtext: _propTypes.default.string
};

var TableWithCustomCells = function TableWithCustomCells() {
  return _react.default.createElement(_Table.default, null, _react.default.createElement(_Table.default.Header, null, _react.default.createElement(_Table.default.HeaderCell, {
    content: "Name",
    key: "NAME",
    minWidth: "small"
  }), _react.default.createElement(_Table.default.HeaderCell, {
    content: "Address",
    key: "ADDRESS",
    minWidth: "medium"
  }), _react.default.createElement(_Table.default.HeaderCell, {
    content: "Phone Number",
    key: "PHONE_NUMBER",
    minWidth: "large"
  })), _react.default.createElement(_Table.default.SingleSelectableRows, null, _react.default.createElement(_Table.default.Row, {
    isSelected: true,
    key: "PERSON_0"
  }, _react.default.createElement(_Table.default.Cell, {
    content: _react.default.createElement(CustomCell, {
      text: "John Smith",
      subtext: "Super Cool Person"
    }),
    key: "NAME"
  }), _react.default.createElement(_Table.default.Cell, {
    content: _react.default.createElement(CustomCell, {
      text: "123 Adams Drive",
      subtext: "Missouri"
    }),
    key: "ADDRESS"
  }), _react.default.createElement(_Table.default.Cell, {
    content: _react.default.createElement(CustomCell, {
      text: "111-222-3333",
      subtext: "Home"
    }),
    key: "PHONE_NUMBER"
  })), _react.default.createElement(_Table.default.Row, {
    key: "PERSON_1"
  }, _react.default.createElement(_Table.default.Cell, {
    content: _react.default.createElement(CustomCell, {
      text: "Jane Smith",
      subtext: "Super Cool Person"
    }),
    key: "NAME"
  }), _react.default.createElement(_Table.default.Cell, {
    content: _react.default.createElement(CustomCell, {
      text: "321 Drive Street",
      subtext: "Kansas"
    }),
    key: "ADDRESS"
  }), _react.default.createElement(_Table.default.Cell, {
    content: _react.default.createElement(CustomCell, {
      text: "111-222-3333",
      subtext: "Cell"
    }),
    key: "PHONE_NUMBER"
  })), _react.default.createElement(_Table.default.Row, {
    key: "PERSON_2"
  }, _react.default.createElement(_Table.default.Cell, {
    content: _react.default.createElement(CustomCell, {
      text: "Dave Smith",
      subtext: "Not Super Cool At All"
    }),
    key: "NAME"
  }), _react.default.createElement(_Table.default.Cell, {
    content: _react.default.createElement(CustomCell, {
      text: "213 Raymond Road",
      subtext: "Alaska"
    }),
    key: "ADDRESS"
  }), _react.default.createElement(_Table.default.Cell, {
    content: _react.default.createElement(CustomCell, {
      text: "111-222-3333",
      subtext: "Work"
    }),
    key: "PHONE_NUMBER"
  }))));
};

var _default = TableWithCustomCells;
exports.default = _default;
