// terra-table/lib/terra-dev-site/test/table/selectable-table/SelectableTableRowsAndSubheaders.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _terraTable = _interopRequireDefault(require("terra-table"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SelectableTableRowsExample = function SelectableTableRowsExample() {
  return _react.default.createElement(_terraTable.default, {
    isStriped: true,
    className: "Table-Custom"
  }, _react.default.createElement(_terraTable.default.Header, {
    className: "Header-Custom"
  }, _react.default.createElement(_terraTable.default.HeaderCell, {
    content: "Name",
    key: "NAME",
    minWidth: "small",
    className: "HeaderCell-Custom"
  }), _react.default.createElement(_terraTable.default.HeaderCell, {
    content: "Address",
    key: "ADDRESS",
    minWidth: "medium",
    sort: "asc"
  }), _react.default.createElement(_terraTable.default.HeaderCell, {
    content: "Phone Number",
    key: "PHONE_NUMBER",
    minWidth: "large"
  })), _react.default.createElement(_terraTable.default.SelectableTableRows, {
    className: "SelectableRows-Custom"
  }, _react.default.createElement(_terraTable.default.Subheader, {
    key: "SUBHEADER_0",
    content: "Single"
  }), _react.default.createElement(_terraTable.default.Row, {
    key: "PERSON_0"
  }, _react.default.createElement(_terraTable.default.Cell, {
    content: "John Smith",
    key: "NAME",
    className: "Cell-Custom"
  }), _react.default.createElement(_terraTable.default.Cell, {
    content: "123 Adams Drive",
    key: "ADDRESS"
  }), _react.default.createElement(_terraTable.default.Cell, {
    content: "111-222-3333",
    key: "PHONE_NUMBER"
  })), _react.default.createElement(_terraTable.default.Row, {
    key: "PERSON_1"
  }, _react.default.createElement(_terraTable.default.Cell, {
    content: "Jane Smith",
    key: "NAME"
  }), _react.default.createElement(_terraTable.default.Cell, {
    content: "321 Drive Street",
    key: "ADDRESS"
  }), _react.default.createElement(_terraTable.default.Cell, {
    content: "111-222-3333",
    key: "PHONE_NUMBER"
  })), _react.default.createElement(_terraTable.default.Subheader, {
    key: "SUBHEADER_1",
    content: "Married",
    className: "Subheader-Custom"
  }), _react.default.createElement(_terraTable.default.Row, {
    key: "PERSON_2"
  }, _react.default.createElement(_terraTable.default.Cell, {
    content: "Dave Smith",
    key: "NAME"
  }), _react.default.createElement(_terraTable.default.Cell, {
    content: "213 Raymond Road",
    key: "ADDRESS"
  }), _react.default.createElement(_terraTable.default.Cell, {
    content: "111-222-3333",
    key: "PHONE_NUMBER"
  })), _react.default.createElement(_terraTable.default.Row, {
    key: "PERSON_3"
  }, _react.default.createElement(_terraTable.default.Cell, {
    content: "Mike Smith",
    key: "NAME"
  }), _react.default.createElement(_terraTable.default.Cell, {
    content: "132 Rock Road",
    key: "ADDRESS"
  }), _react.default.createElement(_terraTable.default.Cell, {
    content: "111-222-3333",
    key: "PHONE_NUMBER"
  }))));
};

var _default = SelectableTableRowsExample;
exports.default = _default;
